### PR TITLE
Add github workflow

### PR DESCRIPTION
**Pull request by kogisin**
closes: #2

### CODE REVIEW

File: .github/workflow/test.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      - name: test & coverage report creation
+        run: make test-cover
+      - name: filter out DONTCOVER
+        run: |
+          excludelist="$(find ./ -type f -name '*.go' | xargs grep -l 'DONTCOVER')"
+          excludelist+=" $(find ./ -type f -name '*.pb.go')"
+          excludelist+=" $(find ./ -type f -name '*.pb.gw.go')"
+          excludelist+=" $(find ./ -type f -path './tests/mocks/*.go')"
+          for filename in ${excludelist}; do
+            filename=$(echo $filename | sed 's/^./github.com\/tendermint\/liquidity/g')
```
Comment (dongsam) [inline .github/workflow/test.yml:57]:
```suggestion
            filename=$(echo $filename | sed 's/^./github.com\/tendermint\/farming/g')
```

I think we need to fix it from liquidity to farming.

File: .github/workflow/test.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      - name: test & coverage report creation
+        run: make test-race
+      - name: filter out DONTCOVER
+        run: |
+          excludelist="$(find ./ -type f -name '*.go' | xargs grep -l 'DONTCOVER')"
+          excludelist+=" $(find ./ -type f -name '*.pb.go')"
+          excludelist+=" $(find ./ -type f -name '*.pb.gw.go')"
+          excludelist+=" $(find ./ -type f -path './tests/mocks/*.go')"
+          for filename in ${excludelist}; do
+            filename=$(echo $filename | sed 's/^./github.com\/tendermint\/liquidity/g')
```
Comment (dongsam) [inline .github/workflow/test.yml:84]:
```suggestion
            filename=$(echo $filename | sed 's/^./github.com\/tendermint\/farming/g')
```

File: .gitignore [hunk 3]
```diff
@@ -1,24 +1,47 @@
...
+dist
+tools-stamp
+proto-tools-stamp
+buf-stamp
 
-# Output of the go coverage tool, specifically when used with LiteIDE
-*.out
+# Data - ideally these don't exist
+app/data/*
+x/liquidity/client/lcd/keys/*
```
Comment (dongsam) [inline .gitignore:28]:
```suggestion
x/farming/client/lcd/keys/*
```

**Review by dongsam — CHANGES_REQUESTED** [review]

File: .github/workflow/test.yml (hunk 1 above)
Comment (kogisin) [reply .github/workflow/test.yml:57]: Oops. I missed this. I'll fix it now.

**Pull request merged**